app/services/sync_service.rb

```diff
@@ -12,45 +12,45 @@ def initialize(host, user_id, access_token, facility_uuid)
     @facility_uuid = facility_uuid
   end
 
-  def sync(request_key, records, request_payload, report_errors_on_class: nil)
+  def sync_all(patients_to_sync)
+    visits_to_sync = Visit.where(patient: patients_to_sync).select(&:unsynced?)
     begin
-      request = to_request(request_key, records, request_payload)
-      response = api_post("api/v2/#{request_key.to_s}/sync", Hash[request_key.to_sym, request])
+      sync('patients', patients_to_sync, SyncPatientPayload)
+      sync('blood_pressures', visits_to_sync, SyncBloodPressurePayload)
+      sync('medical_histories', patients_to_sync, SyncMedicalHistoryPayload)
+      sync('appointments', patients_to_sync, SyncAppointmentPayload)
+      sync('prescription_drugs', patients_to_sync, SyncPrescriptionDrugPayload)
+    rescue
+      raise SyncError, "Error while syncing to Simple Server!"
```

Comment:
We can also add the rescued error message to the sync error string for helping with debugging.

Reply:
I tried doing that, but it was throwing CookieOverflow Exception during development because the error contained all the code inside html.

Will certainly be good to have. Let's pair on this later.